lib/rack/cache/entitystore.rb

```diff
@@ -336,6 +336,38 @@ def self.resolve(uri)
     GAECACHE = GAEStore
     GAE = GAEStore
 
+    # Dummy Entity Store that does not persist entities
+    class Dummy < EntityStore
+      def exist?(key)
+        true
+      end
+
+      def read(key)
+        nil
+      end
+
+      def open(key)
+        nil
```

Comment:
I've finally decided that you're right and it's better to return ```''``` from ```Noop#read``` and ```[]``` from ```Noop#open```. This brings it in line with other backends, which always return objects suitable as Rack bodies from ```#open``` (an empty array can be a Rack body, ```nil``` can't because it doesn't respond to ```each```).

This unfortunately requires a change to ```MetaStore#store``` so that the line
```
response.body = entity_store.open(digest) || response.body
```
is not executed if the entitystore backend is a Noop.

The commit with this change is forthcoming.